GH-1898 - ParquetStore validates snapshot folder does not already exist

benj-fletch opened the pull request:
Enhancement to `ParquetStore#setLatestSnapshot()` to ensure that the snapshot folder being set to does not already exist

m55624 commented:
Hi @benj-fletch, thanks very much for the contribution.  I have just noticed the Travis-CI build has failed on your test, this will need to be fixed before it can be merged, thanks!

benj-fletch commented:
thanks @m55624, I am just resolving the issue; apologies I was a little overzealous in opening the PR!

m55624 commented:
Hi @benj-fletch, just had a read quickly and the ticket is to check the snapshot path does exist, however your fix currently checks it does not exist.  Just on another note you need to use an if round the exists method, this will currently fail on an IOException which is not how you want to test the existence of the directory.  Let me know if you need anymore direction and I can add some change suggestions.

benj-fletch commented:
Hi @m55624, I have updated the PR with a change that actually fixes the ticket. I should have read it more carefully the first time.

I've also added an extra positive test for the method

m55624 commented:
@benj-fletch thanks very much for that, looks great.  Just one small issue with the IOException catch then it should be good to go.  Thanks again for the contribution :)

benj-fletch commented on store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java:
```diff
@@ -405,7 +405,15 @@ public long getLatestSnapshot() throws StoreException {
+                throw new StoreException(String.format("Failed setting currentSnapshot: '%s' does not exist", snapshotPath.toString()));
+            }
+        } catch (final IOException e) {
+            throw new StoreException(String.format("Failed setting currentSnapshot: '%s' cannot be found", getDataDir()), e);
```
Yep, have pushed a change for it just now

benj-fletch commented:
@m55624 No worries, I have made the change. Thanks for taking the time to look at it. I am hoping to contribute more in future and will have a look at other tickets once this one is resolved.

gaffer01 commented:
@benj-fletch Thanks for your contribution.
